components/script/dom/gamepadlist.rs

```diff
@@ -46,6 +44,10 @@ impl GamepadList {
             }
         }
     }
+
+    pub fn remove_gamepad(&self, index: usize) {
+        self.list.borrow_mut().remove(index);
```

Comment:
Let's also  do step 6: "While navigator.[[[gamepads]]](https://www.w3.org/TR/gamepad/#dfn-gamepads) [is not empty](https://infra.spec.whatwg.org/#list-is-empty) and the last [item](https://infra.spec.whatwg.org/#list-item) of navigator.[[[gamepads]]](https://www.w3.org/TR/gamepad/#dfn-gamepads) is null, [remove](https://infra.spec.whatwg.org/#list-remove) the last [item](https://infra.spec.whatwg.org/#list-item) of navigator.[[[gamepads]]](https://www.w3.org/TR/gamepad/#dfn-gamepads)."